Solutions/Azure Active Directory/Analytic Rules/AccountCreatedDeletedByNonApprovedUser.yaml

```diff
@@ -26,18 +26,18 @@ query: |
   AuditLogs
   | where OperationName == "Add user" or OperationName == "Delete user"
   | where Result == "success"
-  | extend InitiatingUser = tostring(parse_json(tostring(InitiatedBy.user)).userPrincipalName)
+  | extend InitiatingUser = tostring(InitiatedBy.user.userPrincipalName)
   | where InitiatingUser has_any (nonapproved_users)
   | project-reorder TimeGenerated, ResourceId, OperationName, InitiatingUser, TargetResources
-  | extend AccountCustomEntity = InitiatingUser, IPCustomEntity = tostring(parse_json(tostring(InitiatedBy.user)).ipAddress)
+  | extend InitiatedUserIpAddress = tostring(InitiatedBy.user.ipAddress)
 entityMappings:
   - entityType: Account
     fieldMappings:
       - identifier: FullName
-        columnName: AccountCustomEntity
+        columnName: InitiatingUser
```

Comment:
Fullname is not a strong entity identifier for Account (its actually a legacy mapping) where we have a UPN we should split in into Name and UPN suffix and have both as entities - https://learn.microsoft.com/en-us/azure/sentinel/entities-reference

Reply:
Modified.
@petebryan Since we are mapping User details here, will it be worthwhile to remove the rows where InitiatedBy does not contain any users? Or should we consider app details in that case?

Reply:
Modifying